src/Refinement/RefinementTransferOperators.jl

```diff
@@ -0,0 +1,166 @@
+
+  # Choose integration space
+  Ω  = best_target(Ω_from,Ω_to)
+  dΩ = Measure(Ω,qdegree)
+  U  = (Ω === Ω_from) ? from : to
+  V  = U.space
+  vh_to = get_fe_basis(to)
+  vh = change_domain(vh_to,Ω)
+
+  # Prepare system
+  sysmat, sysvec = assemble_lhs(Π,Ω_to,to,to.space,qdegree)
+  assem  = SparseMatrixAssembler(to,to.space)
+
+  cache = sysmat, sysvec, Π, assem, Ω, dΩ, U, V, vh
+  return RefinementTransferOperator(eltype(sysmat),from,to,cache)
+end
+
+# Solves the problem Π(uh,vh)_to = Π(uh_from,vh)_Ω for all vh in Vh_to
+function LinearAlgebra.mul!(y,A::RefinementTransferOperator,x)
+  sysmat, sysvec, Π, assem, Ω, dΩ, U, V , vh_Ω = A.caches
+  Ω_to = get_triangulation(A.to)
+
+  # Bring uh to the integration domain
+  uh_from = FEFunction(A.from,x)
+  uh_Ω    = change_domain(uh_from,Ω)
+
+  # Assemble rhs vector
+  contr   = Π(uh_Ω,vh_Ω,dΩ)
+  if Ω !== Ω_to
+    contr = merge_contr_cells(contr,Ω,Ω_to)
+  end
+  vecdata = collect_cell_vector(A.to.space,contr)
+  assemble_vector_add!(sysvec,assem,vecdata)
+
+  # Solve projection
+  IterativeSolvers.cg!(y,sysmat,sysvec)
+  return y
+end
+
+function Base.size(A::RefinementTransferOperator)
+  (num_free_dofs(A.to),num_free_dofs(A.from))
+end
+
+function Base.size(A::RefinementTransferOperator,i::Int)
+  if i == 1
+    return num_free_dofs(A.to)
+  elseif i == 2
+    return num_free_dofs(A.from)
+  else
+    return nothing
+  end
+end
+
+function Base.display(op::RefinementTransferOperator{T,A,B,C}) where {T,A,B,C}
+  s = size(op)
+  println("$(s[1])x$(s[2])  RefinementTransferOperator{$(T)}")
+end
+
+function assemble_lhs(Π,Ω,Uh,Vh,qdegree)
+  dΩ = Measure(Ω,qdegree)
+  uh_dir = FEFunction(Uh,zero_free_values(Uh),get_dirichlet_dof_values(Uh))
+  a(u,v) = Π(u,v,dΩ)
+  b(v)   = a(uh_dir,v)
+
+  sysmat = assemble_matrix(a,Uh,Vh)
+  sysvec = assemble_vector(b,Vh)
+  return sysmat, -sysvec
+end
+
+function merge_contr_cells(a::DomainContribution,rtrian::RefinedTriangulation,ctrian)
+  b = DomainContribution()
+  for trian in get_domains(a)
+    cell_vec = get_contribution(a,trian)
+    res = f2c_cell_contrs(rtrian,cell_vec)
+    add_contribution!(b,ctrian,res)
+  end
+  return b
+end
+
+function f2c_cell_contrs(trian::RefinedTriangulation{Dc,Dp},cell_vec) where {Dc,Dp}
+  @check num_cells(trian) == length(cell_vec)
+
+  model = get_refined_model(trian)
+  glue  = get_glue(model)
+  nF    = num_cells(trian)
+  nC    = num_cells(get_parent(model))
+  nChildren = 4 # Num fine cells per coarse cell
+
+  # Invert fcell_to_ccell
+  fcell_to_ccell = glue.f2c_faces_map[Dc+1]
+  ccell_to_fcell = [fill(-1,nChildren) for i in 1:nC]
+  cidx = fill(1,nC)
+  for iF in 1:nF
+    iC = fcell_to_ccell[iF]
+    ccell_to_fcell[iC][cidx[iC]] = iF
+    cidx[iC] += 1
+  end
+
+  # Map that sums fine contributions for each coarse cell
+  return lazy_map((I,V)->sum(V[I]),ccell_to_fcell,Fill(cell_vec,nC))
+end
+
+
+"""
+"""
+struct RefinementTransferMap{A<:FESpace,B<:FESpace,C<:RefinementTransferOperator} <: Map
```

Comment:
Why is this a `Map`? 

AFAIK, `Map` is typically used in combination with `lazy_map(::Map,...)`.

is this because you want to leverage the `return_cache` and `evaluate!` methods? I do not see the point, but perhaps I am missing something.

Reply:
I agree this type as it is has no purpose. My idea was to have two types: One that works on vectors of dofs and a second one (or the same) that works on `FEFunctions`. The idea is that the `FEFunction` one could be lazy, so that it works well with `change_domain`. I'm open to suggestions.